In [2]:
import pandas as pd
import numpy as np
import matplotlib as plt
import seaborn as sns

In [3]:
%matplotlib inline
pd.set_option('display.max_columns', None)

In [4]:
# import excel file November2012.xlsx
tn_2010_g=pd.read_excel('../data/2010_election/November2010.xls',
                      usecols=['COUNTY','Precinct Name','OFFICENAME','ELECTDATE','BName1','Tally1','BName2','Tally2',
                               'BName3','Tally3','BName4','Tally4','BName5','Tally5','BName6','Tally6','BName7','Tally7',
                               'BName8','Tally8','BName9','Tally9','BName10','Tally10'])
tn_2010_g

COUNTY  Precinct Name                                   OFFICENAME  \
0      ANDERSON  Andersonville                                     Governor   
1      ANDERSON  Andersonville                                     Governor   
2      ANDERSON  Andersonville                     Constitutional Amendment   
3      ANDERSON  Andersonville      U. S. House of Representatives District   
4      ANDERSON  Andersonville                    Tennessee Senate District   
...         ...            ...                                          ...   
12492    WILSON  25 Maple Hill                                     Governor   
12493    WILSON  25 Maple Hill                     Constitutional Amendment   
12494    WILSON  25 Maple Hill      U. S. House of Representatives District   
12495    WILSON  25 Maple Hill                    Tennessee Senate District   
12496    WILSON  25 Maple Hill  Tennessee House of Representatives District   

              ELECTDATE                    BName1  Tally1  \
0      November 2, 2010         . Bill Haslam - R     717   
1      November 2, 2010  . Donald Ray McFolin - I       0   
2      November 2, 2010                     . Yes     813   
3      November 2, 2010   . Chuck Fleischmann - R     519   
4      November 2, 2010       . Randy McNally - R     742   
...                 ...                       ...     ...   
12492  November 2, 2010  . Donald Ray McFolin - I       0   
12493  November 2, 2010                     . Yes    1207   
12494  November 2, 2010          . David Hall - R     963   
12495  November 2, 2010         . Mae Beavers - R     979   
12496  November 2, 2010          . Linda Elam - R     863   

                      BName2  Tally2                     BName3  Tally3  \
0       . Mike McWherter - D     164       . Bayron Binkley - I       1   
1      . Linda Kay Perry - I       0         . James Reesor - I       3   
2                       . No      46                        NaN       0   
3           . John Wolfe - D     141          . Don Barkman - I       2   
4                        NaN       0                        NaN       0   
...                      ...     ...                        ...     ...   
12492  . Linda Kay Perry - I       1         . James Reesor - I       0   
12493                   . No     108                        NaN       0   
12494       . Jim Cooper - D     408  . Stephen W. Collings - I       4   
12495  . George McDonald - D     398                        NaN       0   
12496      . Allen Barry - D     295    . Luther C. Lenning - I      92   

                      BName4  Tally4                    BName5  Tally5  \
0        . Brandon Dodds - I       8   . Samuel David Duck - I       0   
1      . Thomas Smith II - I       2   . Howard M. Switzer - I       1   
2                        NaN       0                       NaN       0   
3           . Mark DeVol - I     190  . Gregory C. Goodwin - I       3   
4                        NaN       0                       NaN       0   
...                      ...     ...                       ...     ...   
12492  . Thomas Smith II - I       1   . Howard M. Switzer - I       2   
12493                    NaN       0                       NaN       0   
12494       . Bill Crook - I       2   . John P. Miglietta - I       1   
12495                    NaN       0                       NaN       0   
12496    . Heather Scott - I     110                       NaN       0   

                                BName6  Tally6                  BName7  \
0                 . David Gatchell - I       0      . June Griffin - I   
1      . Carl Twofeathers Whitaker - I       8  . Write-in - James Lee   
2                                  NaN       0                     NaN   
3               . Robert Humphries - I       1           . Mo Kiah - I   
4                                  NaN       0                     NaN   
...                                ...     ...                     ...   
12492  . Carl Twofe

In [5]:
tn_2010_g.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12497 entries, 0 to 12496
Data columns (total 24 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   COUNTY         12497 non-null  object
 1   Precinct Name  12497 non-null  object
 2   OFFICENAME     12497 non-null  object
 3   ELECTDATE      12497 non-null  object
 4   BName1         12497 non-null  object
 5   Tally1         12497 non-null  int64 
 6   BName2         11339 non-null  object
 7   Tally2         12497 non-null  int64 
 8   BName3         7213 non-null   object
 9   Tally3         12497 non-null  int64 
 10  BName4         6483 non-null   object
 11  Tally4         12497 non-null  int64 
 12  BName5         5992 non-null   object
 13  Tally5         12497 non-null  int64 
 14  BName6         5716 non-null   object
 15  Tally6         12497 non-null  int64 
 16  BName7         5163 non-null   object
 17  Tally7         12497 non-null  int64 
 18  BName8         4891 non-nu

In [6]:
# merge entries in all RNAME columns into new column "name"
tn_2010_g_namemelt=tn_2010_g.melt(id_vars=['COUNTY','Precinct Name','OFFICENAME','ELECTDATE'],value_vars=['BName1','BName2','BName3','BName4','BName5','BName6','BName7','BName8','BName9','BName10'],value_name="candidate")
# drop column "variable" created by .melt
tn_2010_g_namemelt=tn_2010_g_namemelt.drop(['variable'],axis=1)                          
tn_2010_g_namemelt.head()

COUNTY  Precinct Name                               OFFICENAME  \
0  ANDERSON  Andersonville                                 Governor   
1  ANDERSON  Andersonville                                 Governor   
2  ANDERSON  Andersonville                 Constitutional Amendment   
3  ANDERSON  Andersonville  U. S. House of Representatives District   
4  ANDERSON  Andersonville                Tennessee Senate District   

          ELECTDATE                 candidate  
0  November 2, 2010         . Bill Haslam - R  
1  November 2, 2010  . Donald Ray McFolin - I  
2  November 2, 2010                     . Yes  
3  November 2, 2010   . Chuck Fleischmann - R  
4  November 2, 2010       . Randy McNally - R

In [7]:
# merge entries in all TALLY columns into new column "vote_count"
tn_2010_g_pvtallymelt=tn_2010_g.melt(id_vars=['COUNTY','Precinct Name','OFFICENAME','ELECTDATE'],value_vars=['Tally1','Tally2','Tally3','Tally4','Tally5','Tally6','Tally7','Tally8','Tally9','Tally10'],value_name="vote_count")
# drop column "variable" created by .melt
tn_2010_g_pvtallymelt=tn_2010_g_pvtallymelt.drop(['variable'],axis=1)
tn_2010_g_pvtallymelt.head()

COUNTY  Precinct Name                               OFFICENAME  \
0  ANDERSON  Andersonville                                 Governor   
1  ANDERSON  Andersonville                                 Governor   
2  ANDERSON  Andersonville                 Constitutional Amendment   
3  ANDERSON  Andersonville  U. S. House of Representatives District   
4  ANDERSON  Andersonville                Tennessee Senate District   

          ELECTDATE  vote_count  
0  November 2, 2010         717  
1  November 2, 2010           0  
2  November 2, 2010         813  
3  November 2, 2010         519  
4  November 2, 2010         742

In [8]:
# add "vote_count" to tn_2016_namemelt
tally=tn_2010_g_pvtallymelt['vote_count']
tn_2010_g_namemelt=tn_2010_g_namemelt.join(tally)
tn_2010_g_namemelt.head(3)

COUNTY  Precinct Name                OFFICENAME         ELECTDATE  \
0  ANDERSON  Andersonville                  Governor  November 2, 2010   
1  ANDERSON  Andersonville                  Governor  November 2, 2010   
2  ANDERSON  Andersonville  Constitutional Amendment  November 2, 2010   

                  candidate  vote_count  
0         . Bill Haslam - R         717  
1  . Donald Ray McFolin - I           0  
2                     . Yes         813

In [9]:
tn_2010_g_namemelt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 124970 entries, 0 to 124969
Data columns (total 6 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   COUNTY         124970 non-null  object
 1   Precinct Name  124970 non-null  object
 2   OFFICENAME     124970 non-null  object
 3   ELECTDATE      124970 non-null  object
 4   candidate      66327 non-null   object
 5   vote_count     124970 non-null  int64 
dtypes: int64(1), object(5)
memory usage: 5.7+ MB


In [10]:
# lower case column names
tn_2010_g_namemelt.columns=tn_2010_g_namemelt.columns.str.lower()
tn_2010_g_namemelt.head(2)

county  precinct name officename         electdate  \
0  ANDERSON  Andersonville   Governor  November 2, 2010   
1  ANDERSON  Andersonville   Governor  November 2, 2010   

                  candidate  vote_count  
0         . Bill Haslam - R         717  
1  . Donald Ray McFolin - I           0

In [11]:
tn_2010_g_namemelt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 124970 entries, 0 to 124969
Data columns (total 6 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   county         124970 non-null  object
 1   precinct name  124970 non-null  object
 2   officename     124970 non-null  object
 3   electdate      124970 non-null  object
 4   candidate      66327 non-null   object
 5   vote_count     124970 non-null  int64 
dtypes: int64(1), object(5)
memory usage: 5.7+ MB


#### 66327 non-null names  out of 124970 enties. Confirm all are 0 votes and drop rows with no candidate listed

In [12]:
# Create dataframe "blank_candidate_2010" of null valuea in "candidate" of tn_2010_g_namemelt
blank_candidate_2010=tn_2010_g_namemelt[tn_2010_g_namemelt['candidate'].isna()]
blank_candidate_2010

county      precinct name  \
12501   ANDERSON      Andersonville   
12507   ANDERSON         Briceville   
12513   ANDERSON  Blockhouse Valley   
12519   ANDERSON            Clinton   
12525   ANDERSON       Clinton High   
...          ...                ...   
124963    WILSON        25 Leeville   
124965    WILSON      25 Maple Hill   
124966    WILSON      25 Maple Hill   
124968    WILSON      25 Maple Hill   
124969    WILSON      25 Maple Hill   

                                         officename         electdate  \
12501                     Tennessee Senate District  November 2, 2010   
12507                     Tennessee Senate District  November 2, 2010   
12513                     Tennessee Senate District  November 2, 2010   
12519                     Tennessee Senate District  November 2, 2010   
12525                     Tennessee Senate District  November 2, 2010   
...                                             ...               ...   
124963  Tennessee House of Representatives District  November 2, 2010   
124965                                     Governor  November 2, 2010   
124966                     Constitutional Amendment  November 2, 2010   
124968                    Tennessee Senate District  November 2, 2010   
124969  Tennessee House of Representatives District  November 2, 2010   

       candidate  vote_count  
12501        NaN           0  
12507        NaN           0  
12513        NaN           0  
12519        NaN           0  
12525        NaN           0  
...          ...         ...  
124963       NaN           0  
124965       NaN           0  
124966       NaN           0  
124968       NaN           0  
124969       NaN           0  

[58643 rows x 6 columns]

In [14]:
# confirm all are 0 counts
blank_candidate_2010['vote_count'].value_counts()

0    58643
Name: vote_count, dtype: int64

In [15]:
# drop all null candidates
tn_2010_g_namemelt=tn_2010_g_namemelt.dropna().reset_index(drop=True)
tn_2010_g_namemelt

county  precinct name                               officename  \
0      ANDERSON  Andersonville                                 Governor   
1      ANDERSON  Andersonville                                 Governor   
2      ANDERSON  Andersonville                 Constitutional Amendment   
3      ANDERSON  Andersonville  U. S. House of Representatives District   
4      ANDERSON  Andersonville                Tennessee Senate District   
...         ...            ...                                      ...   
66322    WILSON   24 First UMC  U. S. House of Representatives District   
66323    WILSON    25 Leeville                                 Governor   
66324    WILSON    25 Leeville  U. S. House of Representatives District   
66325    WILSON  25 Maple Hill                                 Governor   
66326    WILSON  25 Maple Hill  U. S. House of Representatives District   

              electdate                    candidate  vote_count  
0      November 2, 2010            . Bill Haslam - R         717  
1      November 2, 2010     . Donald Ray McFolin - I           0  
2      November 2, 2010                        . Yes         813  
3      November 2, 2010      . Chuck Fleischmann - R         519  
4      November 2, 2010          . Randy McNally - R         742  
...                 ...                          ...         ...  
66322  November 2, 2010  . James G. Whitfield II - I           3  
66323  November 2, 2010        . Boyce T. McCall - I           1  
66324  November 2, 2010  . James G. Whitfield II - I           2  
66325  November 2, 2010        . Boyce T. McCall - I           0  
66326  November 2, 2010  . James G. Whitfield II - I           4  

[66327 rows x 6 columns]

In [16]:
# confirn all null candidates dropped
tn_2010_g_namemelt[tn_2010_g_namemelt['candidate'].isna()].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 0 entries
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   county         0 non-null      object
 1   precinct name  0 non-null      object
 2   officename     0 non-null      object
 3   electdate      0 non-null      object
 4   candidate      0 non-null      object
 5   vote_count     0 non-null      int64 
dtypes: int64(1), object(5)
memory usage: 0.0+ bytes


In [17]:
tn_2010_g_namemelt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66327 entries, 0 to 66326
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   county         66327 non-null  object
 1   precinct name  66327 non-null  object
 2   officename     66327 non-null  object
 3   electdate      66327 non-null  object
 4   candidate      66327 non-null  object
 5   vote_count     66327 non-null  int64 
dtypes: int64(1), object(5)
memory usage: 3.0+ MB


### Look at cadidates with 0 votes, drop

In [18]:
tn_2010_g_namemelt_0votes=tn_2010_g_namemelt[tn_2010_g_namemelt['vote_count'] == 0]
tn_2010_g_namemelt_0votes

county       precinct name                               officename  \
1      ANDERSON       Andersonville                                 Governor   
7      ANDERSON          Briceville                                 Governor   
13     ANDERSON   Blockhouse Valley                                 Governor   
19     ANDERSON             Clinton                                 Governor   
25     ANDERSON        Clinton High                                 Governor   
...         ...                 ...                                      ...   
66315    WILSON  21 Water Authority                                 Governor   
66316    WILSON  21 Water Authority  U. S. House of Representatives District   
66319    WILSON       23 West Hills                                 Governor   
66321    WILSON        24 First UMC                                 Governor   
66325    WILSON       25 Maple Hill                                 Governor   

              electdate                    candidate  vote_count  
1      November 2, 2010     . Donald Ray McFolin - I           0  
7      November 2, 2010     . Donald Ray McFolin - I           0  
13     November 2, 2010     . Donald Ray McFolin - I           0  
19     November 2, 2010     . Donald Ray McFolin - I           0  
25     November 2, 2010     . Donald Ray McFolin - I           0  
...                 ...                          ...         ...  
66315  November 2, 2010        . Boyce T. McCall - I           0  
66316  November 2, 2010  . James G. Whitfield II - I           0  
66319  November 2, 2010        . Boyce T. McCall - I           0  
66321  November 2, 2010        . Boyce T. McCall - I           0  
66325  November 2, 2010        . Boyce T. McCall - I           0  

[26751 rows x 6 columns]

In [19]:
# drop rows with 0 in "vote_count" column
tn_2010_g_namemelt=tn_2010_g_namemelt[tn_2010_g_namemelt['vote_count'] != 0].reset_index(drop=True)
tn_2010_g_namemelt

county  precinct name                                   officename  \
0      ANDERSON  Andersonville                                     Governor   
1      ANDERSON  Andersonville                     Constitutional Amendment   
2      ANDERSON  Andersonville      U. S. House of Representatives District   
3      ANDERSON  Andersonville                    Tennessee Senate District   
4      ANDERSON  Andersonville  Tennessee House of Representatives District   
...         ...            ...                                          ...   
39571    WILSON  23 West Hills      U. S. House of Representatives District   
39572    WILSON   24 First UMC      U. S. House of Representatives District   
39573    WILSON    25 Leeville                                     Governor   
39574    WILSON    25 Leeville      U. S. House of Representatives District   
39575    WILSON  25 Maple Hill      U. S. House of Representatives District   

              electdate                    candidate  vote_count  
0      November 2, 2010            . Bill Haslam - R         717  
1      November 2, 2010                        . Yes         813  
2      November 2, 2010      . Chuck Fleischmann - R         519  
3      November 2, 2010          . Randy McNally - R         742  
4      November 2, 2010         . Kelly Keisling - R         645  
...                 ...                          ...         ...  
39571  November 2, 2010  . James G. Whitfield II - I           2  
39572  November 2, 2010  . James G. Whitfield II - I           3  
39573  November 2, 2010        . Boyce T. McCall - I           1  
39574  November 2, 2010  . James G. Whitfield II - I           2  
39575  November 2, 2010  . James G. Whitfield II - I           4  

[39576 rows x 6 columns]

In [20]:
tn_2010_g_namemelt['vote_count'].value_counts()

1       8125
2       3990
3       2348
4       1457
5        942
        ... 
1639       1
1511       1
1415       1
1383       1
1983       1
Name: vote_count, Length: 1739, dtype: int64

In [22]:
# group county precincts into county per candidate, in descending order of votes within county
tn_2010_g_int=tn_2010_g_namemelt.groupby(['electdate','officename','county','candidate'], as_index=False)['vote_count'].sum()
tn_2010_g_int=tn_2010_g_int.sort_values(by=['county','vote_count'],ascending=False).reset_index(drop=True)
tn_2010_g_int.head(30)

electdate                                   officename  county  \
0   November 2, 2010                     Constitutional Amendment  WILSON   
1   November 2, 2010                                     Governor  WILSON   
2   November 2, 2010                    Tennessee Senate District  WILSON   
3   November 2, 2010      U. S. House of Representatives District  WILSON   
4   November 2, 2010  Tennessee House of Representatives District  WILSON   
5   November 2, 2010                    Tennessee Senate District  WILSON   
6   November 2, 2010      U. S. House of Representatives District  WILSON   
7   November 2, 2010                                     Governor  WILSON   
8   November 2, 2010  Tennessee House of Representatives District  WILSON   
9   November 2, 2010  Tennessee House of Representatives District  WILSON   
10  November 2, 2010  Tennessee House of Representatives District  WILSON   
11  November 2, 2010      U. S. House of Representatives District  WILSON   
12  November 2, 2010                     Constitutional Amendment  WILSON   
13  November 2, 2010  Tennessee House of Representatives District  WILSON   
14  November 2, 2010      U. S. House of Representatives District  WILSON   
15  November 2, 2010  Tennessee House of Representatives District  WILSON   
16  November 2, 2010                                     Governor  WILSON   
17  November 2, 2010      U. S. House of Representatives District  WILSON   
18  November 2, 2010                                     Governor  WILSON   
19  November 2, 2010                                     Governor  WILSON   
20  November 2, 2010      U. S. House of Representatives District  WILSON   
21  November 2, 2010      U. S. House of Representatives District  WILSON   
22  November 2, 2010      U. S. House of Representatives District  WILSON   
23  November 2, 2010      U. S. House of Representatives District  WILSON   
24  November 2, 2010      U. S. House of Representatives District  WILSON   
25  November 2, 2010      U. S. House of Representatives District  WILSON   
26  November 2, 2010      U. S. House of Representatives District  WILSON   
27  November 2, 2010                                     Governor  WILSON   
28  November 2, 2010      U. S. House of Representatives District  WILSON   
29  November 2, 2010                                     Governor  WILSON   

                          candidate  vote_count  
0                             . Yes       29936  
1                 . Bill Haslam - R       25217  
2                 . Mae Beavers - R       22532  
3                  . David Hall - R       17882  
4                  . Linda Elam - R       12861  
5             . George McDonald - D       10761  
6                  . Jim Cooper - D        9023  
7              . Mike McWherter - D        8724  
8                   . Mark Pody - R        6561  
9               . Stratton Bone - D        5147  
10                . Allen Barry - D        4480  
11                . Diane Black - R        4221  
12                             . No        2496  
13              . Heather Scott - I        2314  
14               . Brett Carter - D        2214  
15          . Luther C. Lenning - I        2197  
16  . Carl Twofeathers Whitaker - I         167  
17              . Jackie Miller - I         125  
18             . Bayron Binkley - I         118  
19              . Brandon Dodds - I         109  
20        . Stephen W. Collings - I         106  
21                   . Jim Boyd - I          95  
22      . John "Big John" Smith - I          71  
23                 . Bill Crook - I          68  
24              . David Purcell - I          63  
25      . James G. Whitfield II - I          58  
26               . Tommy N. Hay - I          58  
27               . June Griffin - I          53  
28            . Brandon E. Gore - I          51  
29            . Linda Kay Perry - I          50

In [23]:
# Capitalize couty names
tn_2010_g_int['county']=tn_2010_g_int['county'].str.title()
tn_2010_g_int.head(5)

electdate                                   officename  county  \
0  November 2, 2010                     Constitutional Amendment  Wilson   
1  November 2, 2010                                     Governor  Wilson   
2  November 2, 2010                    Tennessee Senate District  Wilson   
3  November 2, 2010      U. S. House of Representatives District  Wilson   
4  November 2, 2010  Tennessee House of Representatives District  Wilson   

           candidate  vote_count  
0              . Yes       29936  
1  . Bill Haslam - R       25217  
2  . Mae Beavers - R       22532  
3   . David Hall - R       17882  
4   . Linda Elam - R       12861

In [24]:
tn_2010_g_int.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2606 entries, 0 to 2605
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   electdate   2606 non-null   object
 1   officename  2606 non-null   object
 2   county      2606 non-null   object
 3   candidate   2606 non-null   object
 4   vote_count  2606 non-null   int64 
dtypes: int64(1), object(4)
memory usage: 101.9+ KB


In [25]:
tn_2010_g_int['officename'].unique()

array(['Constitutional Amendment', 'Governor',
       'Tennessee Senate District',
       'U. S. House of Representatives District',
       'Tennessee House of Representatives District'], dtype=object)

### Clean up names and greate separate party column

In [48]:
# new dataframe new_names of non-write-in candidates
new_names=tn_2010_g_int[tn_2010_g_int['candidate'].str.contains("Write")==False]
new_names

electdate                                   officename    county  \
0     November 2, 2010                     Constitutional Amendment    Wilson   
1     November 2, 2010                                     Governor    Wilson   
2     November 2, 2010                    Tennessee Senate District    Wilson   
3     November 2, 2010      U. S. House of Representatives District    Wilson   
4     November 2, 2010  Tennessee House of Representatives District    Wilson   
...                ...                                          ...       ...   
2601  November 2, 2010                                     Governor  Anderson   
2602  November 2, 2010                                     Governor  Anderson   
2603  November 2, 2010                                     Governor  Anderson   
2604  November 2, 2010                                     Governor  Anderson   
2605  November 2, 2010                                     Governor  Anderson   

                     candidate  vote_count  
0                        . Yes       29936  
1            . Bill Haslam - R       25217  
2            . Mae Beavers - R       22532  
3             . David Hall - R       17882  
4             . Linda Elam - R       12861  
...                        ...         ...  
2601      . David Gatchell - I          10  
2602        . Toni K. Hall - I          10  
2603     . Boyce T. McCall - I           7  
2604          . Mike Knois - I           6  
2605  . Donald Ray McFolin - I           3  

[2580 rows x 5 columns]

In [49]:
#drop period in front of names
new_names['candidate']=new_names['candidate'].str[1:]
new_names

/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


electdate                                   officename    county  \
0     November 2, 2010                     Constitutional Amendment    Wilson   
1     November 2, 2010                                     Governor    Wilson   
2     November 2, 2010                    Tennessee Senate District    Wilson   
3     November 2, 2010      U. S. House of Representatives District    Wilson   
4     November 2, 2010  Tennessee House of Representatives District    Wilson   
...                ...                                          ...       ...   
2601  November 2, 2010                                     Governor  Anderson   
2602  November 2, 2010                                     Governor  Anderson   
2603  November 2, 2010                                     Governor  Anderson   
2604  November 2, 2010                                     Governor  Anderson   
2605  November 2, 2010                                     Governor  Anderson   

                    candidate  vote_count  
0                         Yes       29936  
1             Bill Haslam - R       25217  
2             Mae Beavers - R       22532  
3              David Hall - R       17882  
4              Linda Elam - R       12861  
...                       ...         ...  
2601       David Gatchell - I          10  
2602         Toni K. Hall - I          10  
2603      Boyce T. McCall - I           7  
2604           Mike Knois - I           6  
2605   Donald Ray McFolin - I           3  

[2580 rows x 5 columns]

In [50]:
# split name fromm party desination
new_names=new_names['candidate'].str.split(pat='-',expand=True)
new_names

0     1     2
0                      Yes  None  None
1             Bill Haslam      R  None
2             Mae Beavers      R  None
3              David Hall      R  None
4              Linda Elam      R  None
...                    ...   ...   ...
2601       David Gatchell      I  None
2602         Toni K. Hall      I  None
2603      Boyce T. McCall      I  None
2604           Mike Knois      I  None
2605   Donald Ray McFolin      I  None

[2580 rows x 3 columns]

In [54]:
new_names.rename(columns={0:'name',1:'party'},inplace=True)
new_names=new_names.drop(columns=[2])
new_names

name party
0                      Yes  None
1             Bill Haslam      R
2             Mae Beavers      R
3              David Hall      R
4              Linda Elam      R
...                    ...   ...
2601       David Gatchell      I
2602         Toni K. Hall      I
2603      Boyce T. McCall      I
2604           Mike Knois      I
2605   Donald Ray McFolin      I

[2580 rows x 2 columns]

In [55]:
# new dataframe tn_2012_int_wi of only write-in candidates
tn_2010_g_int_wi=tn_2010_g_int[tn_2010_g_int['candidate'].str.contains("Write")==True]
tn_2010_g_int_wi

electdate                                   officename  \
58    November 2, 2010                    Tennessee Senate District   
76    November 2, 2010                                     Governor   
178   November 2, 2010                                     Governor   
206   November 2, 2010                                     Governor   
365   November 2, 2010                                     Governor   
394   November 2, 2010                                     Governor   
410   November 2, 2010  Tennessee House of Representatives District   
418   November 2, 2010      U. S. House of Representatives District   
619   November 2, 2010                                     Governor   
1068  November 2, 2010                                     Governor   
1150  November 2, 2010  Tennessee House of Representatives District   
1181  November 2, 2010                                     Governor   
1375  November 2, 2010                                     Governor   
1469  November 2, 2010  Tennessee House of Representatives District   
1500  November 2, 2010  Tennessee House of Representatives District   
1539  November 2, 2010  Tennessee House of Representatives District   
1556  November 2, 2010  Tennessee House of Representatives District   
1558  November 2, 2010      U. S. House of Representatives District   
1733  November 2, 2010                                     Governor   
2122  November 2, 2010                    Tennessee Senate District   
2123  November 2, 2010                                     Governor   
2124  November 2, 2010                    Tennessee Senate District   
2125  November 2, 2010  Tennessee House of Representatives District   
2177  November 2, 2010                                     Governor   
2204  November 2, 2010                                     Governor   
2331  November 2, 2010                                     Governor   

          county                                 candidate  vote_count  
58    Williamson               . Write-in - Melanie Hollis         274  
76    Williamson               . Write-in - Basil Marceaux           2  
178   Washington               . Write-in - Basil Marceaux           2  
206       Warren               . Write-in - Basil Marceaux           1  
365       Sumner               . Write-in - Basil Marceaux           1  
394     Sullivan               . Write-in - Basil Marceaux           2  
410      Stewart                   . Write-in - James Hart           7  
418      Stewart                   . Write-in - James Hart           2  
619   Rutherford               . Write-in - Basil Marceaux           7  
1068       Maury               . Write-in - Basil Marceaux           1  
1150     Madison          . Write-in - Timothy B. McDonald           2  
1181       Macon               . Write-in - Basil Marceaux           1  
1375        Knox               . Write-in - Basil Marceaux          12  
1469   Humphreys         . Write-in - James E. Travis, Jr.          76  
1500     Houston         . Write-in - James E. Travis, Jr.           4  
1539     Hickman              . Write-in - Teresa Christie           2  
1556       Henry                   . Write-in - James Hart           8  
1558       Henry                   . Write-in - James Hart           7  
1733    Hamilton               . Write-in - Basil Marceaux          14  
2122    Davidson               . Write-in - Melanie Hollis          14  
2123    Davidson               . Write-in - Basil Marceaux          13  
2124    Davidson  . Write-in - Aaron Lee Childress "Skeet"           6  
2125    Davidson                 . Write-in - Chris Polsen           1  
2177    Crockett               . Write-in - Basil Marceaux           1  
2204      Coffee               . Write-in - Basil Marceaux           1  
2331    Cheatham                    . Write-in - James Lee           3

In [56]:
# split name from 'Write-in'
new_wi_name=tn_2010_g_int_wi['candidate'].str.split(pat='-',n=2,expand=True)
new_wi_name

0    1                             2
58    . Write  in                 Melanie Hollis
76    . Write  in                 Basil Marceaux
178   . Write  in                 Basil Marceaux
206   . Write  in                 Basil Marceaux
365   . Write  in                 Basil Marceaux
394   . Write  in                 Basil Marceaux
410   . Write  in                     James Hart
418   . Write  in                     James Hart
619   . Write  in                 Basil Marceaux
1068  . Write  in                 Basil Marceaux
1150  . Write  in            Timothy B. McDonald
1181  . Write  in                 Basil Marceaux
1375  . Write  in                 Basil Marceaux
1469  . Write  in           James E. Travis, Jr.
1500  . Write  in           James E. Travis, Jr.
1539  . Write  in                Teresa Christie
1556  . Write  in                     James Hart
1558  . Write  in                     James Hart
1733  . Write  in                 Basil Marceaux
2122  . Write  in                 Melanie Hollis
2123  . Write  in                 Basil Marceaux
2124  . Write  in    Aaron Lee Childress "Skeet"
2125  . Write  in                   Chris Polsen
2177  . Write  in                 Basil Marceaux
2204  . Write  in                 Basil Marceaux
2331  . Write  in                      James Lee

In [57]:
# create new 'party' columnn with "Write-In"
new_wi_name['party']=new_wi_name[[0,1]].agg('-'.join,axis=1).str[2:]
new_wi_name

0    1                             2      party
58    . Write  in                 Melanie Hollis  Write-in 
76    . Write  in                 Basil Marceaux  Write-in 
178   . Write  in                 Basil Marceaux  Write-in 
206   . Write  in                 Basil Marceaux  Write-in 
365   . Write  in                 Basil Marceaux  Write-in 
394   . Write  in                 Basil Marceaux  Write-in 
410   . Write  in                     James Hart  Write-in 
418   . Write  in                     James Hart  Write-in 
619   . Write  in                 Basil Marceaux  Write-in 
1068  . Write  in                 Basil Marceaux  Write-in 
1150  . Write  in            Timothy B. McDonald  Write-in 
1181  . Write  in                 Basil Marceaux  Write-in 
1375  . Write  in                 Basil Marceaux  Write-in 
1469  . Write  in           James E. Travis, Jr.  Write-in 
1500  . Write  in           James E. Travis, Jr.  Write-in 
1539  . Write  in                Teresa Christie  Write-in 
1556  . Write  in                     James Hart  Write-in 
1558  . Write  in                     James Hart  Write-in 
1733  . Write  in                 Basil Marceaux  Write-in 
2122  . Write  in                 Melanie Hollis  Write-in 
2123  . Write  in                 Basil Marceaux  Write-in 
2124  . Write  in    Aaron Lee Childress "Skeet"  Write-in 
2125  . Write  in                   Chris Polsen  Write-in 
2177  . Write  in                 Basil Marceaux  Write-in 
2204  . Write  in                 Basil Marceaux  Write-in 
2331  . Write  in                      James Lee  Write-in

In [58]:
#drop columns 0 and 1, rename column 2 to"candidate"
new_wi_name=new_wi_name.drop([0,1],axis=1)
new_wi_name

2      party
58                  Melanie Hollis  Write-in 
76                  Basil Marceaux  Write-in 
178                 Basil Marceaux  Write-in 
206                 Basil Marceaux  Write-in 
365                 Basil Marceaux  Write-in 
394                 Basil Marceaux  Write-in 
410                     James Hart  Write-in 
418                     James Hart  Write-in 
619                 Basil Marceaux  Write-in 
1068                Basil Marceaux  Write-in 
1150           Timothy B. McDonald  Write-in 
1181                Basil Marceaux  Write-in 
1375                Basil Marceaux  Write-in 
1469          James E. Travis, Jr.  Write-in 
1500          James E. Travis, Jr.  Write-in 
1539               Teresa Christie  Write-in 
1556                    James Hart  Write-in 
1558                    James Hart  Write-in 
1733                Basil Marceaux  Write-in 
2122                Melanie Hollis  Write-in 
2123                Basil Marceaux  Write-in 
2124   Aaron Lee Childress "Skeet"  Write-in 
2125                  Chris Polsen  Write-in 
2177                Basil Marceaux  Write-in 
2204                Basil Marceaux  Write-in 
2331                     James Lee  Write-in

In [59]:
new_wi_name.rename(columns={2:'name'},inplace=True)
new_wi_name

name      party
58                  Melanie Hollis  Write-in 
76                  Basil Marceaux  Write-in 
178                 Basil Marceaux  Write-in 
206                 Basil Marceaux  Write-in 
365                 Basil Marceaux  Write-in 
394                 Basil Marceaux  Write-in 
410                     James Hart  Write-in 
418                     James Hart  Write-in 
619                 Basil Marceaux  Write-in 
1068                Basil Marceaux  Write-in 
1150           Timothy B. McDonald  Write-in 
1181                Basil Marceaux  Write-in 
1375                Basil Marceaux  Write-in 
1469          James E. Travis, Jr.  Write-in 
1500          James E. Travis, Jr.  Write-in 
1539               Teresa Christie  Write-in 
1556                    James Hart  Write-in 
1558                    James Hart  Write-in 
1733                Basil Marceaux  Write-in 
2122                Melanie Hollis  Write-in 
2123                Basil Marceaux  Write-in 
2124   Aaron Lee Childress "Skeet"  Write-in 
2125                  Chris Polsen  Write-in 
2177                Basil Marceaux  Write-in 
2204                Basil Marceaux  Write-in 
2331                     James Lee  Write-in

In [60]:
# join new_names and new_wi_names
frames=[new_names,new_wi_name]
cleaned_names=pd.concat(frames)
cleaned_names

name      party
0                              Yes       None
1                     Bill Haslam           R
2                     Mae Beavers           R
3                      David Hall           R
4                      Linda Elam           R
...                            ...        ...
2124   Aaron Lee Childress "Skeet"  Write-in 
2125                  Chris Polsen  Write-in 
2177                Basil Marceaux  Write-in 
2204                Basil Marceaux  Write-in 
2331                     James Lee  Write-in 

[2606 rows x 2 columns]

### Join cleaned names and party column to tn_2012_int

In [62]:
name=cleaned_names['name']
party=cleaned_names['party']
tn_2010g_final=tn_2010_g_int.join(name).join(party)
tn_2010g_final

electdate                                   officename    county  \
0     November 2, 2010                     Constitutional Amendment    Wilson   
1     November 2, 2010                                     Governor    Wilson   
2     November 2, 2010                    Tennessee Senate District    Wilson   
3     November 2, 2010      U. S. House of Representatives District    Wilson   
4     November 2, 2010  Tennessee House of Representatives District    Wilson   
...                ...                                          ...       ...   
2601  November 2, 2010                                     Governor  Anderson   
2602  November 2, 2010                                     Governor  Anderson   
2603  November 2, 2010                                     Governor  Anderson   
2604  November 2, 2010                                     Governor  Anderson   
2605  November 2, 2010                                     Governor  Anderson   

                     candidate  vote_count                  name party  
0                        . Yes       29936                   Yes  None  
1            . Bill Haslam - R       25217          Bill Haslam      R  
2            . Mae Beavers - R       22532          Mae Beavers      R  
3             . David Hall - R       17882           David Hall      R  
4             . Linda Elam - R       12861           Linda Elam      R  
...                        ...         ...                   ...   ...  
2601      . David Gatchell - I          10       David Gatchell      I  
2602        . Toni K. Hall - I          10         Toni K. Hall      I  
2603     . Boyce T. McCall - I           7      Boyce T. McCall      I  
2604          . Mike Knois - I           6           Mike Knois      I  
2605  . Donald Ray McFolin - I           3   Donald Ray McFolin      I  

[2606 rows x 7 columns]

In [63]:
tn_2010g_final['party'].value_counts()

 I           1640
 R            388
 D            357
Write-in       26
Colbeck         3
Name: party, dtype: int64

In [65]:
# change "(R)" to "Republican" and "(D)"to "Democratic" in "party" column
tn_2010g_final["party"]=tn_2010g_final['party'].str.replace('R',"Republican").str.replace('D',"Democratic")
tn_2010g_final

electdate                                   officename    county  \
0     November 2, 2010                     Constitutional Amendment    Wilson   
1     November 2, 2010                                     Governor    Wilson   
2     November 2, 2010                    Tennessee Senate District    Wilson   
3     November 2, 2010      U. S. House of Representatives District    Wilson   
4     November 2, 2010  Tennessee House of Representatives District    Wilson   
...                ...                                          ...       ...   
2601  November 2, 2010                                     Governor  Anderson   
2602  November 2, 2010                                     Governor  Anderson   
2603  November 2, 2010                                     Governor  Anderson   
2604  November 2, 2010                                     Governor  Anderson   
2605  November 2, 2010                                     Governor  Anderson   

                     candidate  vote_count                  name        party  
0                        . Yes       29936                   Yes         None  
1            . Bill Haslam - R       25217          Bill Haslam    Republican  
2            . Mae Beavers - R       22532          Mae Beavers    Republican  
3             . David Hall - R       17882           David Hall    Republican  
4             . Linda Elam - R       12861           Linda Elam    Republican  
...                        ...         ...                   ...          ...  
2601      . David Gatchell - I          10       David Gatchell             I  
2602        . Toni K. Hall - I          10         Toni K. Hall             I  
2603     . Boyce T. McCall - I           7      Boyce T. McCall             I  
2604          . Mike Knois - I           6           Mike Knois             I  
2605  . Donald Ray McFolin - I           3   Donald Ray McFolin             I  

[2606 rows x 7 columns]

In [68]:
# create data frame tn_2012_pres for presidental election
tn_2010_gov=tn_2010g_final[tn_2010g_final['officename'].str.match('Governor')]
tn_2010_gov=tn_2010_gov.reset_index(drop=True)
tn_2010_gov

electdate officename    county                        candidate  \
0     November 2, 2010   Governor    Wilson                . Bill Haslam - R   
1     November 2, 2010   Governor    Wilson             . Mike McWherter - D   
2     November 2, 2010   Governor    Wilson  . Carl Twofeathers Whitaker - I   
3     November 2, 2010   Governor    Wilson             . Bayron Binkley - I   
4     November 2, 2010   Governor    Wilson              . Brandon Dodds - I   
...                ...        ...       ...                              ...   
1454  November 2, 2010   Governor  Anderson             . David Gatchell - I   
1455  November 2, 2010   Governor  Anderson               . Toni K. Hall - I   
1456  November 2, 2010   Governor  Anderson            . Boyce T. McCall - I   
1457  November 2, 2010   Governor  Anderson                 . Mike Knois - I   
1458  November 2, 2010   Governor  Anderson         . Donald Ray McFolin - I   

      vote_count                         name        party  
0          25217                 Bill Haslam    Republican  
1           8724              Mike McWherter    Democratic  
2            167   Carl Twofeathers Whitaker             I  
3            118              Bayron Binkley             I  
4            109               Brandon Dodds             I  
...          ...                          ...          ...  
1454          10              David Gatchell             I  
1455          10                Toni K. Hall             I  
1456           7             Boyce T. McCall             I  
1457           6                  Mike Knois             I  
1458           3          Donald Ray McFolin             I  

[1459 rows x 7 columns]

In [72]:
tn_2010_gov.head(20)

electdate officename      county                        candidate  \
0   November 2, 2010   Governor      Wilson                . Bill Haslam - R   
1   November 2, 2010   Governor      Wilson             . Mike McWherter - D   
2   November 2, 2010   Governor      Wilson  . Carl Twofeathers Whitaker - I   
3   November 2, 2010   Governor      Wilson             . Bayron Binkley - I   
4   November 2, 2010   Governor      Wilson              . Brandon Dodds - I   
5   November 2, 2010   Governor      Wilson               . June Griffin - I   
6   November 2, 2010   Governor      Wilson            . Linda Kay Perry - I   
7   November 2, 2010   Governor      Wilson          . Howard M. Switzer - I   
8   November 2, 2010   Governor      Wilson          . Samuel David Duck - I   
9   November 2, 2010   Governor      Wilson               . Toni K. Hall - I   
10  November 2, 2010   Governor      Wilson            . Boyce T. McCall - I   
11  November 2, 2010   Governor      Wilson         . Donald Ray McFolin - I   
12  November 2, 2010   Governor      Wilson               . James Reesor - I   
13  November 2, 2010   Governor      Wilson            . Thomas Smith II - I   
14  November 2, 2010   Governor      Wilson             . David Gatchell - I   
15  November 2, 2010   Governor      Wilson                 . Mike Knois - I   
16  November 2, 2010   Governor  Williamson                . Bill Haslam - R   
17  November 2, 2010   Governor  Williamson             . Mike McWherter - D   
18  November 2, 2010   Governor  Williamson             . Bayron Binkley - I   
19  November 2, 2010   Governor  Williamson  . Carl Twofeathers Whitaker - I   

    vote_count                         name        party  
0        25217                 Bill Haslam    Republican  
1         8724              Mike McWherter    Democratic  
2          167   Carl Twofeathers Whitaker             I  
3          118              Bayron Binkley             I  
4          109               Brandon Dodds             I  
5           53                June Griffin             I  
6           50             Linda Kay Perry             I  
7           40           Howard M. Switzer             I  
8           29           Samuel David Duck             I  
9           22                Toni K. Hall             I  
10          19             Boyce T. McCall             I  
11          14          Donald Ray McFolin             I  
12          13                James Reesor             I  
13          10             Thomas Smith II             I  
14           8              David Gatchell             I  
15           7                  Mike Knois             I  
16       48518                 Bill Haslam    Republican  
17       11085              Mike McWherter    Democratic  
18         397              Bayron Binkley             I  
19         203   Carl Twofeathers Whitaker             I

### Add column that has total vote count by count for use in calculating county vote percentages

In [73]:
# create new dataframe "county_vote" to use to find sum of votes per county
county_vote=tn_2010_gov.loc[0:,['county','vote_count']]
county_vote

county  vote_count
0       Wilson       25217
1       Wilson        8724
2       Wilson         167
3       Wilson         118
4       Wilson         109
...        ...         ...
1454  Anderson          10
1455  Anderson          10
1456  Anderson           7
1457  Anderson           6
1458  Anderson           3

[1459 rows x 2 columns]

In [74]:
#sum county votes
county_vote_sum=county_vote.groupby(['county'])['vote_count'].sum()
county_vote_sum

county
Anderson      19225
Bedford        9796
Benton         5043
Bledsoe        3814
Blount        29963
              ...  
Wayne          3557
Weakley        9449
White          6994
Williamson    60603
Wilson        34600
Name: vote_count, Length: 95, dtype: int64

In [75]:
# use "county_vote_sum" to add column "sum_county_vote" to tn_2016_pres correlating with value in each row
tn_2010_gov['sum_county_vote']=tn_2010_gov['county'].map(county_vote_sum)
tn_2010_gov.head(20)

electdate officename      county                        candidate  \
0   November 2, 2010   Governor      Wilson                . Bill Haslam - R   
1   November 2, 2010   Governor      Wilson             . Mike McWherter - D   
2   November 2, 2010   Governor      Wilson  . Carl Twofeathers Whitaker - I   
3   November 2, 2010   Governor      Wilson             . Bayron Binkley - I   
4   November 2, 2010   Governor      Wilson              . Brandon Dodds - I   
5   November 2, 2010   Governor      Wilson               . June Griffin - I   
6   November 2, 2010   Governor      Wilson            . Linda Kay Perry - I   
7   November 2, 2010   Governor      Wilson          . Howard M. Switzer - I   
8   November 2, 2010   Governor      Wilson          . Samuel David Duck - I   
9   November 2, 2010   Governor      Wilson               . Toni K. Hall - I   
10  November 2, 2010   Governor      Wilson            . Boyce T. McCall - I   
11  November 2, 2010   Governor      Wilson         . Donald Ray McFolin - I   
12  November 2, 2010   Governor      Wilson               . James Reesor - I   
13  November 2, 2010   Governor      Wilson            . Thomas Smith II - I   
14  November 2, 2010   Governor      Wilson             . David Gatchell - I   
15  November 2, 2010   Governor      Wilson                 . Mike Knois - I   
16  November 2, 2010   Governor  Williamson                . Bill Haslam - R   
17  November 2, 2010   Governor  Williamson             . Mike McWherter - D   
18  November 2, 2010   Governor  Williamson             . Bayron Binkley - I   
19  November 2, 2010   Governor  Williamson  . Carl Twofeathers Whitaker - I   

    vote_count                         name        party  sum_county_vote  
0        25217                 Bill Haslam    Republican            34600  
1         8724              Mike McWherter    Democratic            34600  
2          167   Carl Twofeathers Whitaker             I            34600  
3          118              Bayron Binkley             I            34600  
4          109               Brandon Dodds             I            34600  
5           53                June Griffin             I            34600  
6           50             Linda Kay Perry             I            34600  
7           40           Howard M. Switzer             I            34600  
8           29           Samuel David Duck             I            34600  
9           22                Toni K. Hall             I            34600  
10          19             Boyce T. McCall             I            34600  
11          14          Donald Ray McFolin             I            34600  
12          13                James Reesor             I            34600  
13          10             Thomas Smith II             I            34600  
14           8              David Gatchell             I            34600  
15           7                  Mike Knois             I            34600  
16       48518                 Bill Haslam    Republican            60603  
17       11085              Mike McWherter    Democratic            60603  
18         397              Bayron Binkley             I            60603  
19         203   Carl Twofeathers Whitaker             I            60603

In [76]:
# create "pct_county_vote" column for tn_2016_pres
tn_2010_gov['pct_county_vote']=round(tn_2010_gov['vote_count']/tn_2010_gov['sum_county_vote']*100,2)
tn_2010_gov.head(20)

electdate officename      county                        candidate  \
0   November 2, 2010   Governor      Wilson                . Bill Haslam - R   
1   November 2, 2010   Governor      Wilson             . Mike McWherter - D   
2   November 2, 2010   Governor      Wilson  . Carl Twofeathers Whitaker - I   
3   November 2, 2010   Governor      Wilson             . Bayron Binkley - I   
4   November 2, 2010   Governor      Wilson              . Brandon Dodds - I   
5   November 2, 2010   Governor      Wilson               . June Griffin - I   
6   November 2, 2010   Governor      Wilson            . Linda Kay Perry - I   
7   November 2, 2010   Governor      Wilson          . Howard M. Switzer - I   
8   November 2, 2010   Governor      Wilson          . Samuel David Duck - I   
9   November 2, 2010   Governor      Wilson               . Toni K. Hall - I   
10  November 2, 2010   Governor      Wilson            . Boyce T. McCall - I   
11  November 2, 2010   Governor      Wilson         . Donald Ray McFolin - I   
12  November 2, 2010   Governor      Wilson               . James Reesor - I   
13  November 2, 2010   Governor      Wilson            . Thomas Smith II - I   
14  November 2, 2010   Governor      Wilson             . David Gatchell - I   
15  November 2, 2010   Governor      Wilson                 . Mike Knois - I   
16  November 2, 2010   Governor  Williamson                . Bill Haslam - R   
17  November 2, 2010   Governor  Williamson             . Mike McWherter - D   
18  November 2, 2010   Governor  Williamson             . Bayron Binkley - I   
19  November 2, 2010   Governor  Williamson  . Carl Twofeathers Whitaker - I   

    vote_count                         name        party  sum_county_vote  \
0        25217                 Bill Haslam    Republican            34600   
1         8724              Mike McWherter    Democratic            34600   
2          167   Carl Twofeathers Whitaker             I            34600   
3          118              Bayron Binkley             I            34600   
4          109               Brandon Dodds             I            34600   
5           53                June Griffin             I            34600   
6           50             Linda Kay Perry             I            34600   
7           40           Howard M. Switzer             I            34600   
8           29           Samuel David Duck             I            34600   
9           22                Toni K. Hall             I            34600   
10          19             Boyce T. McCall             I            34600   
11          14          Donald Ray McFolin             I            34600   
12          13                James Reesor             I            34600   
13          10             Thomas Smith II             I            34600   
14           8              David Gatchell             I            34600   
15           7                  Mike Knois             I            34600   
16       48518                 Bill Haslam    Republican            60603   
17       11085              Mike McWherter    Democratic            60603   
18         397              Bayron Binkley             I            60603   
19         203   Carl Twofeathers Whitaker             I            60603   

    pct_county_vote  
0             72.88  
1             25.21  
2              0.48  
3              0.34  
4              0.32  
5              0.15  
6              0.14  
7              0.12  
8              0.08  
9              0.06  
10             0.05  
11             0.04  
12             0.04  
13             0.03  
14             0.02  
15             0.02  
16            80.06  
17            18.29  
18             0.66  
19             0.33

In [77]:
# how many parties?
tn_2010_gov['party'].value_counts()

 I             1255
 Republican      95
 Democratic      95
Write-in         14
Name: party, dtype: int64

## Create dataframe "tn_2010_county_gov_pct"  with percent values per party per county

In [78]:
# create new dataframe with reduced columns values only
tn_2010_county_gov_pct=tn_2010_gov.loc[0:,['electdate','county','party','pct_county_vote']]
tn_2010_county_gov_pct

electdate    county        party  pct_county_vote
0     November 2, 2010    Wilson   Republican            72.88
1     November 2, 2010    Wilson   Democratic            25.21
2     November 2, 2010    Wilson            I             0.48
3     November 2, 2010    Wilson            I             0.34
4     November 2, 2010    Wilson            I             0.32
...                ...       ...          ...              ...
1454  November 2, 2010  Anderson            I             0.05
1455  November 2, 2010  Anderson            I             0.05
1456  November 2, 2010  Anderson            I             0.04
1457  November 2, 2010  Anderson            I             0.03
1458  November 2, 2010  Anderson            I             0.02

[1459 rows x 4 columns]

In [79]:
# create new dataframe with Republican values only
tn_2010g_rep_pct=tn_2010_county_gov_pct[tn_2010_county_gov_pct['party']== " Republican"].reset_index(drop=True)
tn_2010g_rep_pct

electdate      county        party  pct_county_vote
0   November 2, 2010      Wilson   Republican            72.88
1   November 2, 2010  Williamson   Republican            80.06
2   November 2, 2010       White   Republican            64.94
3   November 2, 2010     Weakley   Republican            51.76
4   November 2, 2010       Wayne   Republican            73.26
..               ...         ...          ...              ...
90  November 2, 2010      Blount   Republican            79.38
91  November 2, 2010     Bledsoe   Republican            66.26
92  November 2, 2010      Benton   Republican            51.36
93  November 2, 2010     Bedford   Republican            69.50
94  November 2, 2010    Anderson   Republican            72.50

[95 rows x 4 columns]

In [80]:
# rename 'pct_county_vote' to 'rep_pct' and drop "party" column to prepare for merge 
tn_2010g_rep_pct=tn_2010g_rep_pct.rename(columns={'pct_county_vote':'Republican'},inplace=False).drop(['party'], axis=1)
tn_2010g_rep_pct

electdate      county  Republican
0   November 2, 2010      Wilson       72.88
1   November 2, 2010  Williamson       80.06
2   November 2, 2010       White       64.94
3   November 2, 2010     Weakley       51.76
4   November 2, 2010       Wayne       73.26
..               ...         ...         ...
90  November 2, 2010      Blount       79.38
91  November 2, 2010     Bledsoe       66.26
92  November 2, 2010      Benton       51.36
93  November 2, 2010     Bedford       69.50
94  November 2, 2010    Anderson       72.50

[95 rows x 3 columns]

In [81]:
# create new dataframe with Democratic values only
tn_2010g_dem_pct=tn_2010_county_gov_pct[tn_2010_county_gov_pct['party']== " Democratic"].reset_index(drop=True)
tn_2010g_dem_pct

electdate      county        party  pct_county_vote
0   November 2, 2010      Wilson   Democratic            25.21
1   November 2, 2010  Williamson   Democratic            18.29
2   November 2, 2010       White   Democratic            32.31
3   November 2, 2010     Weakley   Democratic            46.49
4   November 2, 2010       Wayne   Democratic            25.13
..               ...         ...          ...              ...
90  November 2, 2010      Blount   Democratic            18.19
91  November 2, 2010     Bledsoe   Democratic            31.83
92  November 2, 2010      Benton   Democratic            47.23
93  November 2, 2010     Bedford   Democratic            28.31
94  November 2, 2010    Anderson   Democratic            24.89

[95 rows x 4 columns]

In [82]:
# rename 'pct_county_vote' to 'dem_pct' and drop "party" to prepare for merge 
tn_2010g_dem_pct=tn_2010g_dem_pct.rename(columns={'pct_county_vote':'Democratic'},inplace=False).drop(['party'], axis=1)
tn_2010g_dem_pct

electdate      county  Democratic
0   November 2, 2010      Wilson       25.21
1   November 2, 2010  Williamson       18.29
2   November 2, 2010       White       32.31
3   November 2, 2010     Weakley       46.49
4   November 2, 2010       Wayne       25.13
..               ...         ...         ...
90  November 2, 2010      Blount       18.19
91  November 2, 2010     Bledsoe       31.83
92  November 2, 2010      Benton       47.23
93  November 2, 2010     Bedford       28.31
94  November 2, 2010    Anderson       24.89

[95 rows x 3 columns]

In [86]:
# create new dataframe with non Republican or Democratic values only
tn_2010g_other_pct=tn_2010_county_gov_pct[(tn_2010_county_gov_pct['party']!=" Republican")&(tn_2010_county_gov_pct['party']!=" Democratic")].reset_index(drop=True)
tn_2010g_other_pct

electdate    county party  pct_county_vote
0     November 2, 2010    Wilson     I             0.48
1     November 2, 2010    Wilson     I             0.34
2     November 2, 2010    Wilson     I             0.32
3     November 2, 2010    Wilson     I             0.15
4     November 2, 2010    Wilson     I             0.14
...                ...       ...   ...              ...
1264  November 2, 2010  Anderson     I             0.05
1265  November 2, 2010  Anderson     I             0.05
1266  November 2, 2010  Anderson     I             0.04
1267  November 2, 2010  Anderson     I             0.03
1268  November 2, 2010  Anderson     I             0.02

[1269 rows x 4 columns]

In [88]:
# sum by county
tn_2010g_other_pct=tn_2010g_other_pct.groupby(['county'],as_index=False)['pct_county_vote'].sum()
tn_2010g_other_pct

county  pct_county_vote
0     Anderson             2.62
1      Bedford             2.19
2       Benton             1.42
3      Bledsoe             1.92
4       Blount             2.41
..         ...              ...
90       Wayne             1.61
91     Weakley             1.71
92       White             2.76
93  Williamson             1.65
94      Wilson             1.89

[95 rows x 2 columns]

In [89]:
# rename 'pct_county_vote' to 'other_pct' and drop "party" to prepare for merge 
tn_2010g_other_pct=tn_2010g_other_pct.rename(columns={'pct_county_vote':'Other'})
tn_2010g_other_pct

county  Other
0     Anderson   2.62
1      Bedford   2.19
2       Benton   1.42
3      Bledsoe   1.92
4       Blount   2.41
..         ...    ...
90       Wayne   1.61
91     Weakley   1.71
92       White   2.76
93  Williamson   1.65
94      Wilson   1.89

[95 rows x 2 columns]

In [90]:
# join tn_201og_rep, tn_2010g_dem_pct and tn_2010g_other_pct to form tn_2010_county_gov_pct
dem=tn_2010g_dem_pct['Democratic']
other=tn_2010g_other_pct['Other']
tn_2010_county_gov_pct=tn_2010g_rep_pct.join(dem).join(other)
tn_2010_county_gov_pct

electdate      county  Republican  Democratic  Other
0   November 2, 2010      Wilson       72.88       25.21   2.62
1   November 2, 2010  Williamson       80.06       18.29   2.19
2   November 2, 2010       White       64.94       32.31   1.42
3   November 2, 2010     Weakley       51.76       46.49   1.92
4   November 2, 2010       Wayne       73.26       25.13   2.41
..               ...         ...         ...         ...    ...
90  November 2, 2010      Blount       79.38       18.19   1.61
91  November 2, 2010     Bledsoe       66.26       31.83   1.71
92  November 2, 2010      Benton       51.36       47.23   2.76
93  November 2, 2010     Bedford       69.50       28.31   1.65
94  November 2, 2010    Anderson       72.50       24.89   1.89

[95 rows x 5 columns]

### Add column for Republican margin over Democrat for heat mapping in Tableau

In [91]:
tn_2010_county_gov_pct['Rep_Margin']=tn_2010_county_gov_pct['Republican']-tn_2010_county_gov_pct['Democratic']
tn_2010_county_gov_pct.head(50)

electdate      county  Republican  Democratic  Other  Rep_Margin
0   November 2, 2010      Wilson       72.88       25.21   2.62       47.67
1   November 2, 2010  Williamson       80.06       18.29   2.19       61.77
2   November 2, 2010       White       64.94       32.31   1.42       32.63
3   November 2, 2010     Weakley       51.76       46.49   1.92        5.27
4   November 2, 2010       Wayne       73.26       25.13   2.41       48.13
5   November 2, 2010  Washington       76.95       21.63   1.61       55.32
6   November 2, 2010      Warren       59.61       37.77   2.47       21.84
7   November 2, 2010   Van Buren       57.29       40.25   2.41       17.04
8   November 2, 2010       Union       75.15       22.83   1.43       52.32
9   November 2, 2010      Unicoi       75.48       22.66   1.59       52.82
10  November 2, 2010   Trousdale       56.06       41.93   1.93       14.13
11  November 2, 2010      Tipton       68.63       29.69   1.82       38.94
12  November 2, 2010      Sumner       73.17       25.30   2.69       47.87
13  November 2, 2010    Sullivan       77.44       21.20   1.95       56.24
14  November 2, 2010     Stewart       54.91       43.05   2.24       11.86
15  November 2, 2010       Smith       58.36       39.67   2.09       18.69
16  November 2, 2010      Shelby       46.74       51.86   1.08       -5.12
17  November 2, 2010      Sevier       84.01       13.63   1.95       70.38
18  November 2, 2010  Sequatchie       70.74       27.37   2.44       43.37
19  November 2, 2010       Scott       74.48       23.41   1.88       51.07
20  November 2, 2010  Rutherford       68.77       29.33   1.96       39.44
21  November 2, 2010   Robertson       67.94       30.28   1.87       37.66
22  November 2, 2010       Roane       73.79       23.74   4.71       50.05
23  November 2, 2010        Rhea       76.06       21.58   1.51       54.48
24  November 2, 2010      Putnam       69.83       28.44   1.53       41.39
25  November 2, 2010        Polk       64.86       32.69   1.93       32.17
26  November 2, 2010     Pickett       66.20       31.66   1.56       34.54
27  November 2, 2010       Perry       55.34       41.83   2.23       13.51
28  November 2, 2010     Overton       58.68       39.24   2.45       19.44
29  November 2, 2010       Obion       53.99       43.21   2.06       10.78
30  November 2, 2010      Morgan       69.81       27.33   2.91       42.48
31  November 2, 2010       Moore       68.59       27.68   1.33       40.91
32  November 2, 2010  Montgomery       61.80       35.87   1.60       25.93
33  November 2, 2010      Monroe       75.02       23.09   2.55       51.93
34  November 2, 2010       Meigs       68.28       29.52   2.28       38.76
35  November 2, 2010     Mcnairy       64.05       34.44   1.92       29.61
36  November 2, 2010      Mcminn       76.63       21.66   1.94       54.97
37  November 2, 2010       Maury       65.50       32.82   1.35       32.68
38  November 2, 2010    Marshall       63.50       34.59   1.85       28.91
39  November 2, 2010      Marion       62.47       35.69   1.45       26.78
40  November 2, 2010     Madison       55.74       43.20   2.31       12.54
41  November 2, 2010       Macon       68.21       29.84   2.34       38.37
42  November 2, 2010      Loudon       82.07       15.78   1.88       66.29
43  November 2, 2010     Lincoln       72.30       24.15   2.44       48.15
44  November 2, 2010       Lewis       63.42       33.97   2.22       29.45
45  November 2, 2010    Lawrence       66.54       32.00   1.79       34.54
46  November 2, 2010  Lauderdale       54.74       43.64   1.87       11.10
47  November 2, 2010        Lake       47.38       49.32   3.31       -1.94
48  November 2, 2010        Knox       81.07       17.06   1.62       64.01
49  November 2, 2010     Johnson       75.59       22.62   1.47       52.97

In [94]:
#check spelling of DeKalb to see if it needs to be corrected
tn_2010_county_gov_pct.tail(50)

electdate      county  Republican  Democratic  Other  Rep_Margin
45  November 2, 2010    Lawrence       66.54       32.00   1.79       34.54
46  November 2, 2010  Lauderdale       54.74       43.64   1.87       11.10
47  November 2, 2010        Lake       47.38       49.32   3.31       -1.94
48  November 2, 2010        Knox       81.07       17.06   1.62       64.01
49  November 2, 2010     Johnson       75.59       22.62   1.47       52.97
50  November 2, 2010   Jefferson       79.54       18.24   2.62       61.30
51  November 2, 2010     Jackson       53.02       44.54   3.56        8.48
52  November 2, 2010   Humphreys       53.62       44.51   2.16        9.11
53  November 2, 2010     Houston       47.90       49.74   1.94       -1.84
54  November 2, 2010     Hickman       59.80       37.89   1.05       21.91
55  November 2, 2010       Henry       55.20       43.33   1.83       11.87
56  November 2, 2010   Henderson       69.43       28.73   1.93       40.70
57  November 2, 2010     Haywood       45.70       52.96   1.67       -7.26
58  November 2, 2010     Hawkins       75.29       22.76   1.72       52.53
59  November 2, 2010      Hardin       69.10       28.98   1.50       40.12
60  November 2, 2010    Hardeman       50.45       47.25   2.22        3.20
61  November 2, 2010     Hancock       73.25       24.21   1.89       49.04
62  November 2, 2010    Hamilton       65.59       32.81   2.33       32.78
63  November 2, 2010     Hamblen       79.47       19.19   3.75       60.28
64  November 2, 2010      Grundy       55.07       42.03   2.87       13.04
65  November 2, 2010      Greene       76.83       21.11   2.79       55.72
66  November 2, 2010    Grainger       74.79       22.75   2.08       52.04
67  November 2, 2010       Giles       60.25       37.52   2.83       22.73
68  November 2, 2010      Gibson       57.57       40.88   2.14       16.69
69  November 2, 2010    Franklin       63.35       34.73   2.46       28.62
70  November 2, 2010    Fentress       71.44       27.04   1.75       44.40
71  November 2, 2010     Fayette       68.78       29.71   2.33       39.07
72  November 2, 2010        Dyer       66.48       28.79   2.45       37.69
73  November 2, 2010     Dickson       64.09       34.04   1.80       30.05
74  November 2, 2010      Dekalb       58.40       39.62   1.88       18.78
75  November 2, 2010     Decatur       56.31       41.80   2.10       14.51
76  November 2, 2010    Davidson       48.44       49.11   1.90       -0.67
77  November 2, 2010  Cumberland       77.71       20.29   2.37       57.42
78  November 2, 2010    Crockett       64.49       34.42   1.41       30.07
79  November 2, 2010      Coffee       69.43       28.49   1.97       40.94
80  November 2, 2010       Cocke       75.69       22.08   2.04       53.61
81  November 2, 2010        Clay       59.07       38.98   1.37       20.09
82  November 2, 2010   Claiborne       74.45       22.85   1.52       51.60
83  November 2, 2010     Chester       69.39       28.80   1.70       40.59
84  November 2, 2010    Cheatham       68.92       29.16   2.01       39.76
85  November 2, 2010      Carter       76.98       21.43   1.86       55.55
86  November 2, 2010     Carroll       56.72       41.83   2.04       14.89
87  November 2, 2010      Cannon       61.51       36.09   2.45       25.42
88  November 2, 2010    Campbell       72.75       24.78   2.61       47.97
89  November 2, 2010     Bradley       82.10       16.29   1.43       65.81
90  November 2, 2010      Blount       79.38       18.19   1.61       61.19
91  November 2, 2010     Bledsoe       66.26       31.83   1.71       34.43
92  November 2, 2010      Benton       51.36       47.23   2.76        4.13
93  November 2, 2010     Bedford       69.50       28.31   1.65       41.19
94  November 2, 2010    Anderson       72.50       24.89   1.89       47.61

In [97]:
# correct county names: Dekalb to DeKalb, Mcminn to McMinn, Mcnairy to McNairy 
tn_2010_county_gov_pct['county']=tn_2010_county_gov_pct['county'].str.replace('Dekalb','DeKalb').str.replace('Mcminn','McMinn').str.replace('Mcnairy','McNairy')
tn_2010_county_gov_pct.tail(21)

electdate      county  Republican  Democratic  Other  Rep_Margin
74  November 2, 2010      DeKalb       58.40       39.62   1.88       18.78
75  November 2, 2010     Decatur       56.31       41.80   2.10       14.51
76  November 2, 2010    Davidson       48.44       49.11   1.90       -0.67
77  November 2, 2010  Cumberland       77.71       20.29   2.37       57.42
78  November 2, 2010    Crockett       64.49       34.42   1.41       30.07
79  November 2, 2010      Coffee       69.43       28.49   1.97       40.94
80  November 2, 2010       Cocke       75.69       22.08   2.04       53.61
81  November 2, 2010        Clay       59.07       38.98   1.37       20.09
82  November 2, 2010   Claiborne       74.45       22.85   1.52       51.60
83  November 2, 2010     Chester       69.39       28.80   1.70       40.59
84  November 2, 2010    Cheatham       68.92       29.16   2.01       39.76
85  November 2, 2010      Carter       76.98       21.43   1.86       55.55
86  November 2, 2010     Carroll       56.72       41.83   2.04       14.89
87  November 2, 2010      Cannon       61.51       36.09   2.45       25.42
88  November 2, 2010    Campbell       72.75       24.78   2.61       47.97
89  November 2, 2010     Bradley       82.10       16.29   1.43       65.81
90  November 2, 2010      Blount       79.38       18.19   1.61       61.19
91  November 2, 2010     Bledsoe       66.26       31.83   1.71       34.43
92  November 2, 2010      Benton       51.36       47.23   2.76        4.13
93  November 2, 2010     Bedford       69.50       28.31   1.65       41.19
94  November 2, 2010    Anderson       72.50       24.89   1.89       47.61

In [105]:
tn_2010_county_gov_pct.head(37)

electdate      county  Republican  Democratic  Other  Rep_Margin
0   November 2, 2010      Wilson       72.88       25.21   2.62       47.67
1   November 2, 2010  Williamson       80.06       18.29   2.19       61.77
2   November 2, 2010       White       64.94       32.31   1.42       32.63
3   November 2, 2010     Weakley       51.76       46.49   1.92        5.27
4   November 2, 2010       Wayne       73.26       25.13   2.41       48.13
5   November 2, 2010  Washington       76.95       21.63   1.61       55.32
6   November 2, 2010      Warren       59.61       37.77   2.47       21.84
7   November 2, 2010   Van Buren       57.29       40.25   2.41       17.04
8   November 2, 2010       Union       75.15       22.83   1.43       52.32
9   November 2, 2010      Unicoi       75.48       22.66   1.59       52.82
10  November 2, 2010   Trousdale       56.06       41.93   1.93       14.13
11  November 2, 2010      Tipton       68.63       29.69   1.82       38.94
12  November 2, 2010      Sumner       73.17       25.30   2.69       47.87
13  November 2, 2010    Sullivan       77.44       21.20   1.95       56.24
14  November 2, 2010     Stewart       54.91       43.05   2.24       11.86
15  November 2, 2010       Smith       58.36       39.67   2.09       18.69
16  November 2, 2010      Shelby       46.74       51.86   1.08       -5.12
17  November 2, 2010      Sevier       84.01       13.63   1.95       70.38
18  November 2, 2010  Sequatchie       70.74       27.37   2.44       43.37
19  November 2, 2010       Scott       74.48       23.41   1.88       51.07
20  November 2, 2010  Rutherford       68.77       29.33   1.96       39.44
21  November 2, 2010   Robertson       67.94       30.28   1.87       37.66
22  November 2, 2010       Roane       73.79       23.74   4.71       50.05
23  November 2, 2010        Rhea       76.06       21.58   1.51       54.48
24  November 2, 2010      Putnam       69.83       28.44   1.53       41.39
25  November 2, 2010        Polk       64.86       32.69   1.93       32.17
26  November 2, 2010     Pickett       66.20       31.66   1.56       34.54
27  November 2, 2010       Perry       55.34       41.83   2.23       13.51
28  November 2, 2010     Overton       58.68       39.24   2.45       19.44
29  November 2, 2010       Obion       53.99       43.21   2.06       10.78
30  November 2, 2010      Morgan       69.81       27.33   2.91       42.48
31  November 2, 2010       Moore       68.59       27.68   1.33       40.91
32  November 2, 2010  Montgomery       61.80       35.87   1.60       25.93
33  November 2, 2010      Monroe       75.02       23.09   2.55       51.93
34  November 2, 2010       Meigs       68.28       29.52   2.28       38.76
35  November 2, 2010     McNairy       64.05       34.44   1.92       29.61
36  November 2, 2010      McMinn       76.63       21.66   1.94       54.97

In [106]:
#read TN_county_index_shape.csv to add index from TN Counties shape file for joining in Tableau
county_index=pd.read_csv('../data/TN_county_index_shape.csv')
county_index

county  INDEX_NO
0     Anderson        79
1      Bedford        26
2       Benton        41
3      Bledsoe        31
4       Blount        52
..         ...       ...
90       Wayne         6
91     Weakley        56
92       White        67
93  Williamson        44
94      Wilson        65

[95 rows x 2 columns]

In [107]:
# join with tn_2012_county_pres_pct
tn_2010_county_gov_pct=tn_2010_county_gov_pct.merge(county_index,on="county")
tn_2010_county_gov_pct

electdate      county  Republican  Democratic  Other  Rep_Margin  \
0   November 2, 2010      Wilson       72.88       25.21   2.62       47.67   
1   November 2, 2010  Williamson       80.06       18.29   2.19       61.77   
2   November 2, 2010       White       64.94       32.31   1.42       32.63   
3   November 2, 2010     Weakley       51.76       46.49   1.92        5.27   
4   November 2, 2010       Wayne       73.26       25.13   2.41       48.13   
..               ...         ...         ...         ...    ...         ...   
90  November 2, 2010      Blount       79.38       18.19   1.61       61.19   
91  November 2, 2010     Bledsoe       66.26       31.83   1.71       34.43   
92  November 2, 2010      Benton       51.36       47.23   2.76        4.13   
93  November 2, 2010     Bedford       69.50       28.31   1.65       41.19   
94  November 2, 2010    Anderson       72.50       24.89   1.89       47.61   

    INDEX_NO  
0         65  
1         44  
2         67  
3         56  
4          6  
..       ...  
90        52  
91        31  
92        41  
93        26  
94        79  

[95 rows x 7 columns]

In [108]:
tn_2010_county_gov_pct.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 95 entries, 0 to 94
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   electdate   95 non-null     object 
 1   county      95 non-null     object 
 2   Republican  95 non-null     float64
 3   Democratic  95 non-null     float64
 4   Other       95 non-null     float64
 5   Rep_Margin  95 non-null     float64
 6   INDEX_NO    95 non-null     int64  
dtypes: float64(4), int64(1), object(2)
memory usage: 5.9+ KB


In [109]:
# covert "electdate" to datetime from object
tn_2010_county_gov_pct['electdate']=pd.to_datetime(tn_2010_county_gov_pct['electdate'],format='%B %d, %Y')
tn_2010_county_gov_pct

electdate      county  Republican  Democratic  Other  Rep_Margin  INDEX_NO
0  2010-11-02      Wilson       72.88       25.21   2.62       47.67        65
1  2010-11-02  Williamson       80.06       18.29   2.19       61.77        44
2  2010-11-02       White       64.94       32.31   1.42       32.63        67
3  2010-11-02     Weakley       51.76       46.49   1.92        5.27        56
4  2010-11-02       Wayne       73.26       25.13   2.41       48.13         6
..        ...         ...         ...         ...    ...         ...       ...
90 2010-11-02      Blount       79.38       18.19   1.61       61.19        52
91 2010-11-02     Bledsoe       66.26       31.83   1.71       34.43        31
92 2010-11-02      Benton       51.36       47.23   2.76        4.13        41
93 2010-11-02     Bedford       69.50       28.31   1.65       41.19        26
94 2010-11-02    Anderson       72.50       24.89   1.89       47.61        79

[95 rows x 7 columns]

In [110]:
tn_2010_county_gov_pct.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 95 entries, 0 to 94
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   electdate   95 non-null     datetime64[ns]
 1   county      95 non-null     object        
 2   Republican  95 non-null     float64       
 3   Democratic  95 non-null     float64       
 4   Other       95 non-null     float64       
 5   Rep_Margin  95 non-null     float64       
 6   INDEX_NO    95 non-null     int64         
dtypes: datetime64[ns](1), float64(4), int64(1), object(1)
memory usage: 5.9+ KB


In [112]:
tn_2010_county_gov_pct.to_excel(r'../data/2010_election/tn_2010_county_gov_pct.xlsx',index=False)